Actor prepare_upgrade_transaction ends with ERROR and exit code 0

On subscribed machine `leapp ugprade` ended with an "ERROR" in prepare_upgrade_transaction and it stopped processing of upgrade. 
`[ERROR] Actor: prepare_upgrade_transaction Message: Error: Error while trying to execute command: Command '['systemd-nspawn', '--register=no', '-D', '/tmp/leapp-overlay/merged', '--', '/bin/bash', '-c', u"/usr/bin/dnf shell -y --setopt=protected_packages= --disablerepo '*' --releasever 8 --allowerasing --best --nogpgcheck --setopt=tsflags=test --enablerepo rhel-8-for-x86_64-baseos-htb-rpms,rhel-8-for-x86_64-appstream-htb-rpms /var/lib/leapp/dnf-script.txt"]' returned non-zero exit status 1`

Processing of `leapp upgrade` also ended with exit status 0 in case of found error. Reported in https://github.com/oamg/leapp/issues/372

Packages:
leapp-0.4.0-1.201812060905Z.b3ebf82.master.el7_5.noarch
leapp-repository-0.4.0-1.201812060911Z.f3090b8.master.el7_5.noarch

```
# leapp upgrade
...
many rows
...
(218/651): glibc-headers-2.28-38.el8.x86_64.rpm                                                                                                                                    552 kB/s | 464 kB     00:00    
(219/651): kernel-headers-4.18.0-48.el8.x86_64.rpm                                                                                                                                 674 kB/s | 1.4 MB     00:02    
[MIRROR] GConf2-3.2.6-22.el8.x86_64.rpm: Status code: 404 for https://cdn.stage.redhat.com/content/htb/rhel8/8/x86_64/appstream/os/Packages/g/GConf2-3.2.6-22.el8.x86_64.rpm                                      
[FAILED] GConf2-3.2.6-22.el8.x86_64.rpm: No more mirrors to try - All mirrors were already tried without success                                                                                                  
(221-222/651): webkit2gtk3-plugin-process-gtk2-2.22.3-1.el8.x86_64.rpm                 48% [=========================================                                            ] 2.4 MB/s | 272 MB     02:01 ETA
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: Error downloading packages:
  Cannot download Packages/g/GConf2-3.2.6-22.el8.x86_64.rpm: All mirrors were tried
Container merged failed with error code 1.
2018-12-07 10:51:43.311 INFO     PID: 25678 leapp.workflow.Download: Starting stage After of phase Download
2018-12-07 10:51:43.515 INFO     PID: 25678 leapp.workflow: Workflow interrupted due to the FailPhase error policy

============================================================
                        ERRORS
============================================================

2018-12-07 09:51:43.148352 [ERROR] Actor: prepare_upgrade_transaction Message: Error: Error while trying to execute command: Command '['systemd-nspawn', '--register=no', '-D', '/tmp/leapp-overlay/merged', '--', '/bin/bash', '-c', u"/usr/bin/dnf shell -y --setopt=protected_packages= --disablerepo '*' --releasever 8 --allowerasing --best --nogpgcheck --setopt=tsflags=test --enablerepo rhel-8-for-x86_64-baseos-htb-rpms,rhel-8-for-x86_64-appstream-htb-rpms /var/lib/leapp/dnf-script.txt"]' returned non-zero exit status 1

============================================================
                     END OF ERRORS
============================================================

# echo $?
0
```

--- a/leapp/cli/upgrade/__init__.py
+++ b/leapp/cli/upgrade/__init__.py
@@ -125,6 +125,9 @@ def upgrade(args):
 
     report_errors(workflow.errors)
 
+    if workflow.errors:
+        sys.exit(1)
+
 
 @command('list-runs', help='List previous Leapp upgrade executions')
 def list_runs(args):

